controllers/machineset_controller.go

```diff
@@ -639,7 +630,7 @@ func (r *MachineSetReconciler) calculateStatus(ctx context.Context, cluster *clu
 }
 
 // patchMachineSetStatus attempts to update the Status.Replicas of the given MachineSet.
-func (r *MachineSetReconciler) patchMachineSetStatus(ctx context.Context, ms *clusterv1.MachineSet, newStatus *clusterv1.MachineSetStatus) (*clusterv1.MachineSet, error) {
+func (r *MachineSetReconciler) patchMachineSetStatus(ctx context.Context, ms *clusterv1.MachineSet, newStatus *clusterv1.MachineSetStatus) *clusterv1.MachineSet {
```

Comment:
This isn't patching any longer, we should probably rename it to something different?

Reply:
We might also want to consider to merge these last bits into `calculateStatus`